### PR TITLE
Initialize mmp_last_write when the mmp thread starts

### PR DESCRIPTION
**Pull request by ofaaland**
### Motivation and Context

A great deal of time may go by between when mmp_init() is called and
the MMP thread starts, particularly if there are bad devices, because
there is I/O.  If this time is too long,

    (gethrtime() - mmp_last_write) > mmp_fail_ns

at the time the MMP thread starts.  If MMP is configured to suspend
the pool, the pool will be suspended immediately.

This can be see in issue #10838
### Description

The value of mmp_last_write doesn't matter before the mmp thread
starts.  To give the MMP thread time to issue MMP writes and have them
land, initialize mmp_last_write just before the MMP thread starts.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by behlendorf — APPROVED** [review]
Looks good, thanks for running down this corner case!

File: module/zfs/mmp.c [hunk 1]
```diff
@@ -557,6 +549,16 @@ mmp_thread(void *arg)
 
 	mmp_thread_enter(mmp, &cpr);
 
+	/*
```
Comment (dinatale2) [inline module/zfs/mmp.c:552]: nit: Maybe mention in this comment something about why the last write time is set here instead of elsewhere?

Comment (ofaaland) [reply module/zfs/mmp.c:552]: OK, comment updated.

**Review by dinatale2 — APPROVED** [review]

**Comment by ofaaland** [comment]
Hi @behlendorf is there going to be a zfs-0.8.5?  Can you remind me how to get this on that patch list?  Thanks!

**Comment by behlendorf** [comment]
@ofaaland I went ahead and added this to the 0.8-release project so we can track it.  We also created a zfs-0.8.5-staging branch which you can open a PR against to make sure this gets included.

**Comment by homerl** [comment]
Why not let zpool import return failed?  The zpool suspend cause too many production environments crashes.

**Pull request merged**